dsm/resource_app.go

```diff
@@ -189,9 +216,82 @@ func resourceUpdateApp(ctx context.Context, d *schema.ResourceData, m interface{
 			})
 			return diags
 		}
-		return resourceReadApp(ctx, d, m)
 	}
-	return nil
+    //Modified by Ravi Gopal
+	app_object := make(map[string]interface{})
+
+	if d.HasChange("default_group") {
+		if default_group := d.Get("default_group").(string); len(default_group) > 0 {
+			app_object["default_group"] = d.Get("default_group")
+		}
+	}
+	if d.HasChange("other_group") {
+		old_group, new_group := d.GetChange("other_group")
+		/*compares the old and new state
+		 * and seggregtes the new groups and groups that are to be deleted.
+		*/
+		add_group_ids, del_group_ids := form_add_and_del_groups(old_group, new_group)
+		//Add the groups to be deleted
+		if len(del_group_ids) > 0 {
+			app_object["del_groups"] = del_group_ids
+		}
+		//Add the new groups
+		if len(add_group_ids) > 0 {
+			add_group_perms := form_group_permissions(d.Get("other_group_permissions"))
+			app_add_group := make(map[string]interface{})
+			for i := 0; i < len(add_group_ids); i++ {
+				if perms, ok := add_group_perms[add_group_ids[i]]; ok {
+					app_add_group[add_group_ids[i]] = perms
+				}else{
+					app_add_group[add_group_ids[i]] = []string{"SIGN", "VERIFY", "ENCRYPT", "DECRYPT", "WRAPKEY", "UNWRAPKEY", "DERIVEKEY", "MACGENERATE", "MACVERIFY", "EXPORT", "MANAGE", "AGREEKEY", "AUDIT"}
+				}
+			}
+			app_object["add_groups"] = app_add_group
+		}
+	}
+	if d.HasChange("description") {
```

Comment:
Why guard this with a condition? Why not just set it unconditionally?

Reply:
Removed the condition.